File: backend/rxas400adm-system/src/main/java/com/rxas400adm/system/entity/SysDoc.java
package com.rxas400adm.system.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 知识库文档（rx_sys_doc，V66）：纯 DB 元数据，无 IFS 存储。
 */
@Data
@TableName("rx_sys_doc")
public class SysDoc {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String content;

    /** 分类 */
    private String category;

    /** 标签（逗号分隔） */
    private String tags;

    /** DRAFT / PUBLISHED */
    private String status;

    private String createdBy;

    private String updatedBy;

    private LocalDateTime createdTime;

    private LocalDateTime updatedTime;
}

File: frontend/src/api/sysDoc.ts
import request from './request'

/** 知识库文档状态 */
export type SysDocStatus = 'DRAFT' | 'PUBLISHED'

/** 知识库文档 */
export interface SysDocItem {
  id: number
  title: string
  content?: string
  category?: string
  tags?: string
  status: SysDocStatus
  createdBy?: string
  updatedBy?: string
  createdTime?: string
  updatedTime?: string
}

/** 分页查询知识库文档 */
export const listSysDocs = (params: {
  current?: number
  size?: number
  keyword?: string
  status?: string
  category?: string
} = {}): Promise<{ total: number; records: SysDocItem[] }> =>
  request.get('/sys-docs', { params })

/** 获取单个知识库文档 */
export const getSysDoc = (id: number): Promise<SysDocItem> =>
  request.get(`/sys-docs/${id}`)

/** 创建知识库文档 */
export const createSysDoc = (data: {
  title: string
  content?: string
  category?: string
  tags?: string
}) => request.post('/sys-docs', data)

/** 更新知识库文档 */
export const updateSysDoc = (
  id: number,
  data: {
    title: string
    content?: string
    category?: string
    tags?: string
    status?: SysDocStatus
  },
) => request.put(`/sys-docs/${id}`, data)

/** 删除知识库文档 */
export const deleteSysDoc = (id: number) => request.delete(`/sys-docs/${id}`)

/** 切换发布状态 */
export const toggleSysDocStatus = (id: number, status: SysDocStatus) =>
  request.put(`/sys-docs/${id}`, { status })
